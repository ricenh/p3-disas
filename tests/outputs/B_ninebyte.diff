--- outputs/B_ninebyte.txt	2023-11-01 19:33:18.898501438 -0400
+++ expected/B_ninebyte.txt	2023-10-10 11:54:35.000000000 -0400
@@ -1,11 +1,6 @@
-Usage: ../y86 <option(s)> mini-elf-file
- Options are:
-  -h      Display usage
-  -H      Show the Mini-ELF header
-  -a      Show all with brief memory
-  -f      Show all with full memory
-  -s      Show the program headers
-  -m      Show the memory contents (brief)
-  -M      Show the memory contents (full)
-  -d      Disassemble code contents
-  -D      Disassemble data contents
+Disassembly of executable contents:
+  0x100:                               | .pos 0x100 code
+  0x100:                               | _start:
+  0x100: 70 88 77 66 55 44 33 22 11    |   jmp 0x1122334455667788
+  0x109: 80 11 22 33 44 55 66 77 88    |   call 0x8877665544332211
+
