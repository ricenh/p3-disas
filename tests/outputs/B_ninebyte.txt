Disassembly of executable contents:
  0x100:                               | .pos 0x100 code
  0x100:                               | _start:
  0x100: 70 88 77 66 55 44 33 22 11    |   jmp0x109
  0x109: 80 11 22 33 44 55 66 77 88    |   call0x112

