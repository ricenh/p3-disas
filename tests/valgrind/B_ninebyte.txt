==2329383== Memcheck, a memory error detector
==2329383== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==2329383== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==2329383== Command: ../y86 -d inputs/ninebyte.o
==2329383== 
../y86: invalid option -- 'd'
Usage: ../y86 <option(s)> mini-elf-file
 Options are:
  -h      Display usage
  -H      Show the Mini-ELF header
  -a      Show all with brief memory
  -f      Show all with full memory
  -s      Show the program headers
  -m      Show the memory contents (brief)
  -M      Show the memory contents (full)
  -d      Disassemble code contents
  -D      Disassemble data contents
==2329383== 
==2329383== HEAP SUMMARY:
==2329383==     in use at exit: 0 bytes in 0 blocks
==2329383==   total heap usage: 1 allocs, 1 frees, 8,192 bytes allocated
==2329383== 
==2329383== All heap blocks were freed -- no leaks are possible
==2329383== 
==2329383== For lists of detected and suppressed errors, rerun with: -s
==2329383== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
