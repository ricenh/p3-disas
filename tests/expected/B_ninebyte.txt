Disassembly of executable contents:
  0x100:                               | .pos 0x100 code
  0x100:                               | _start:
  0x100: 70 88 77 66 55 44 33 22 11    |   jmp 0x1122334455667788
  0x109: 80 11 22 33 44 55 66 77 88    |   call 0x8877665544332211

